qutebrowser/completion/completer.py

```diff
@@ -121,12 +114,11 @@ def _open_completion_if_needed(self):
         if not config.get('completion', 'auto-open'):
             connected = self._connect_signals(True)
             if connected:
-                self.update_completion()
+                self._update_completion()
 
     def _model(self):
         """Convenience method to get the current completion model."""
-        completion = objreg.get('completion', scope='window',
-                                window=self._win_id)
+        completion = self.parent()
```

Comment:
That's what the tests are for :grinning: 
I think if you set the parent to anything else, the tests will fail and you'll realize what happened as the line `completion = self.parent()` makes the assumption pretty clear.

I did go back and forth on using a separate attribute though, and this option only slightly won out.